examples/Google.Cloud.Functions.Examples.StorageImageAnnotator/Function.cs

```diff
@@ -0,0 +1,139 @@
+        private readonly ILogger _logger;
+
+        /// <summary>
+        ///  Constructor accepting all our dependencies. The clients are configured in
+        ///  Startup.cs.
+        /// </summary>
+        public Function(ImageAnnotatorClient visionClient, StorageClient storageClient, ILogger<Function> logger) =>
+            (_visionClient, _storageClient, _logger) = (visionClient, storageClient, logger);
+
+        /// <summary>
+        /// Entry point for the function. This is called whenever a new storage file is created (including
+        /// when we create one ourselves - we need to be careful not to create an infinite loop!)
+        /// </summary>
+        /// <param name="payload">The storage object that's been uploaded.</param>
+        /// <param name="context">Event context (event ID etc)</param>
+        public async Task HandleAsync(StorageObject payload, Context context)
+        {
+            // Only handle JPEG images. (This prevents us from trying to perform image recognition on
+            // the files we upload.)
+            if (payload.ContentType != JpegContentType || Path.GetExtension(payload.Name) != JpegExtension)
+            {
+                _logger.LogInformation("Ignoring file {name} with content type {contentType}",
+                    payload.Name, payload.ContentType);
+                return;
+            }
+
+            var annotations = await AnnotateImageAsync(payload);
+            var text = DescribeAnnotations(annotations);
+            string newObjectName = await CreateDescriptionFileAsync(payload, text);
+            _logger.LogInformation("Created object {object} with image annotations", newObjectName);
+        }
+
+        /// <summary>
+        /// Use the Vision API to annotate the image. The Vision API is able to annotate Google Cloud Storage
+        /// objects directly based on their name, so we don't need to download the object and then upload it
+        /// to the Vision API.
+        /// </summary>
+        private Task<AnnotateImageResponse> AnnotateImageAsync(StorageObject payload)
+        {
+            var features = new[] { FaceDetection, LandmarkDetection, TextDetection, LogoDetection, LabelDetection }
+                .Select(type => new Feature { Type = type, MaxResults = 20 });
+            var request = new AnnotateImageRequest
+            {
+                Image = Image.FromUri($"gs://{payload.Bucket}/{payload.Name}"),
+                Features = { features }
+            };
+            return _visionClient.AnnotateAsync(request);
+        }
+
+        /// <summary>
+        /// Converts a result from the Google Cloud Vision API into a textual representation.
+        /// This is very primitive, but is sufficient for demonstration purposes.
+        /// An alternative approach would be to create a new image file that highlighted all the faces,
+        /// added labels for all the landmarks, etc. (That would require more care to avoid an infinite loop.)
```

Comment:
ni: I would expand on the infinite loop comment here with something like ... because we would be uploading anothet jpeg file that we ourselves would send for annotation ...

Reply:
Will do.